src/simulations/step.jl

```diff
@@ -38,6 +38,44 @@ function CommonSolve.step!(model::ABM, n::Union{Function, Int} = 1)
     end
 end
 
+"""
+    step!(model::EventQueueABM, t::Real)
+
+Evolve the model executing the scheduled events until the time 
+reaches the current time + `t`.
+"""
+function CommonSolve.step!(model::EventQueueABM, t::Real)
+    model_t = getfield(model, :time)
+    t0 = model_t[]
+    while model_t[] < t0 + t
+        event, t_event = dequeue_pair!(abmqueue(model))
+        process_event!(event, model)
+        model_t[] = t_event
+    end
+end
+
+function process_event!(event, model)
+    id, event_idx = event.id, event.event_index
+    !haskey(agent_container(model), id) && return
+    agent = model[id]
+    agent_type = findfirst(isequal(typeof(agent)), tuple_agenttype(model))
+    event_function! = abmevents(model)[agent_type][event_idx]
+    event_function!(agent, model)
+    !haskey(agent_container(model), id) && return
+    propensities = abmrates(agent, model) .* nagents(model)
```

Comment:
It's not really clear to me why here we multiply by `nagents(model)`, shouldn't it be the number of agents of the related agent type?

Reply:
actually to me is not clear why applying a function such as `nagents` in general? shouldn't the user define his own rates? Given that we let choosing a function then every transformation of the propensities seems not useful. But it's not entirely clear to me how the algorithm should work, so I could be mistaken

Reply:
@MA-Ramirez @Datseris I'd need some help on this if you can give some :-)

Reply:
If nobody has insights on this I will remove it, it is in any case possible to achieve through passing a function in `abmrates `

Reply:
Sorry for the late reply. Yes, you should remove this part and if the user needs to they can call `nagents` in `abmrates`.